conf/up4.bess

```diff
@@ -199,10 +201,48 @@ if ntf:
   _in -> ntf
   _in = ntf
 
-_in -> qerLookup::Qos(fields=[ {'attr_name':'src_iface', 'num_bytes':1}, \
-                               {'attr_name':'qer_id', 'num_bytes':4}, \
-                               {'attr_name':'fseid', 'num_bytes':8}], \
-                             values=[{'attr_name':'qfi', 'num_bytes':1}])
+_in -> qerLookup::Qos(fields=[{'attr_name':'src_iface', 'num_bytes':1}, \
+                              {'attr_name':'qer_id', 'num_bytes':4}, \
+                              {'attr_name':'fseid', 'num_bytes':8}], \
+                      values=[{'attr_name':'qfi', 'num_bytes':1}])
+
+executeFAR::Split(size=1, attribute='action')
+farOrSliceMeter = executeFAR
+if parser.slice_uplink_rate_limit_bytes_per_second is not None or parser.slice_downlink_rate_limit_bytes_per_second is not None:
+    # sliceMeter enforces a per slice, per direction meter rate limit
+    sliceMeter::Qos(fields=[{'attr_name':'src_iface', 'num_bytes':1}])
+    # Reserved gates, reject rule adds with gate=1/2/3
+    m_meter  = 0 # Placeholder gate not connected. Will meter if lookup result returns this gate
+    m_green  = 1 # For green traffic
+    m_yellow = 2 # For yellow traffic
+    m_red    = 3 # For red traffic
+    # User defined gates
+    m_fail   = 4 # For lookup failure traffic
+    # Admit green and yellow, drop red
+    sliceMeter:m_green -> executeFAR
+    sliceMeter:m_yellow -> executeFAR
+    sliceMeter:m_red -> sliceMeterRed::Sink()
+    sliceMeter:m_fail -> sliceMeterLookupFail::Sink()
+    sliceMeter.set_default_gate(gate=m_fail)
+    # Add two static slice meter entries for uplink and downlink.
+    # cir and pir are equal, meaning we have no yellow packets.
+    if parser.slice_uplink_rate_limit_bytes_per_second is not None:
+        sliceMeter.add(fields=[{'value_int': Access}],
+                       gate=m_meter,
+                       cir=parser.slice_uplink_rate_limit_bytes_per_second,
```

Comment:
should set cir to 1

Reply:
I don't understand your comment. Hardcode to 1 byte/sec?

Reply:
oh, you referred to cir vs pir. Got it. This would mark all traffic as yellow, right?

Reply:
right.